test/integration/react-18/test/index.test.js

```diff
@@ -52,54 +60,148 @@ async function getDevOutput(dir) {
-      expect(output).toMatch(USING_CREATE_ROOT)
-      expect(output).toMatch(UNSUPPORTED_PRERELEASE)
+    test('supported version of react in build', async () => {
+      const output = await getBuildOutput(appDir)
+      expect(output).not.toMatch(USING_CREATE_ROOT)
+      expect(output).not.toMatch(UNSUPPORTED_PRERELEASE)
     })
   })
 
-  describe('dev', () => {
-    test('supported version of React', async () => {
-      let output = await getDevOutput(dirSupported)
-      expect(output).not.toMatch(USING_CREATE_ROOT)
-      expect(output).not.toMatch(UNSUPPORTED_PRERELEASE)
+  describe('warns with stable supported version of react-dom', () => {
+    beforeAll(async () => {
+      const reactDomPkgPath = join(
+        appDir,
+        'node_modules/react-dom/package.json'
+      )
+      await fs.outputJson(reactDomPkgPath, {
+        name: 'react-dom',
+        version: '18.0.0-alpha-c76e4dbbc-20210722',
+      })
+    })
+    afterAll(async () => await fs.remove(join(appDir, 'node_modules')))
+
+    test('prerelease version of react in dev', async () => {
+      const output = await getDevOutput(appDir)
+      expect(output).toMatch(USING_CREATE_ROOT)
+      expect(output).toMatch(UNSUPPORTED_PRERELEASE)
     })
 
-    test('prerelease version of React', async () => {
-      let output = await getDevOutput(dirPrerelease)
+    test('prerelease version of react in build', async () => {
+      const output = await getBuildOutput(appDir)
       expect(output).toMatch(USING_CREATE_ROOT)
       expect(output).toMatch(UNSUPPORTED_PRERELEASE)
     })
   })
+})
 
-  describe('hydration', () => {
-    const appDir = join(__dirname, '../prerelease')
-    let app
-    let appPort
-    beforeAll(async () => {
-      await fs.remove(join(appDir, '.next'))
-      await nextBuild(appDir, [dirPrerelease], {
-        nodeArgs,
-        stdout: true,
-        stderr: true,
-      })
-      appPort = await findPort()
-      app = await nextStart(appDir, appPort, { nodeArgs })
+describe('React 18 basics', () => {
+  const context = {}
+  beforeAll(async () => {
+    await fs.remove(join(appDir, '.next'))
+    await nextBuild(appDir, [appDir], { nodeArgs })
+    context.appPort = await findPort()
+    context.app = await nextStart(appDir, context.appPort, { nodeArgs })
+  })
+  afterAll(async () => {
+    await killApp(context.app)
+  })
+
+  it('hydrates correctly for normal page', async () => {
+    const browser = await webdriver(context.appPort, '/')
+    expect(await browser.eval('window.didHydrate')).toBe(true)
+    expect(await browser.elementById('react-dom-version').text()).toMatch(/18/)
+  })
+
+  it('should works with suspense in ssg', async () => {
+    const res1 = await fetchViaHTTP(context.appPort, '/suspense/thrown')
+    const res2 = await fetchViaHTTP(context.appPort, '/suspense/no-thrown')
+
+    expect(res1.status).toBe(200)
+    expect(res2.status).toBe(200)
+  })
+
+  it('should not preload modules on server side', async () => {
```

Comment:
Is the module part of this already covered by the dynamic/concurrent tests? If so, I think it might be clearer to specifically test that Next only renders the fallbacks in React 18 (without concurrent features enabled), and rename the test name/URL

Reply:
Specifically: this is a bit confusing, because modules _should_ be preloaded on the server if you're using `next/dynamic` _without_ suspense. So I had to think about what this was actually testing quite a bit.